operator-common/src/main/java/io/strimzi/operator/common/operator/resource/DeploymentConfigOperator.java

```diff
@@ -72,9 +73,35 @@ protected Integer currentScale(String namespace, String name) {
     private boolean isObserved(String namespace, String name) {
         DeploymentConfig dep = get(namespace, name);
         if (dep != null)   {
-            return dep.getMetadata().getGeneration().equals(dep.getStatus().getObservedGeneration());
+            // Get the roll out status
+            //     => Sometimes it takes OCP some time before the generations are updated.
+            //        So we need to check the conditions in addition to detect such situation.
+            boolean rollOutNotStarting = true;
+            DeploymentCondition progressing = getProgressingCondition(dep);
+
+            if (progressing != null)    {
+                rollOutNotStarting = progressing.getReason() != null && !"Unknown".equals(progressing.getStatus());
```

Comment:
should it be valid just for "Ready". Is it possible that the status is "NotReady" so anyway different from "Unkown"?

Reply:
Not sure I understand the question. When the Reason is null and Status is Unknown, the generation has not been updated yet.

Reply:
So the state is netiehr Ready nor NotReady, but Unknown.

Reply:
Right.